### ISSUE TITLE
[=BUG=] Youtube requires sign in to confirm you are not a bot

### ISSUE THREAD

**Issue by Tapawingo** [play-dl/play-dl#370]
#### Describe the bug

 - Node.js version: `18.14.0`
 - OS & version: `Linux Ubuntu 20.04.5 LTS`

A rejection while attempting to get a youtube stream from a url: "Sign in to confirm you're not a bot". I have tried setting new cookies but the error remains, I have also tried without any cookies to the same result. 

#### Actual behavior

A rejection is thrown.

#### Expected behavior
 
I expect a youtube video stream returned.

#### Code to reproduce

```js
const dl = require('play-dl');

const getSourceStream = async (track) => {
    var readableStream = await dl.stream(track, {
        quality: 0,
        discordPlayerCompatibility: true
      });

      return readableStream;
};

getSourceStream("https://youtu.be/dQw4w9WgXcQ");
```

#### ScreenShots

![image](https://github.com/play-dl/play-dl/assets/60672612/e03bdc65-d0f0-4212-9d12-505137e1f382)

#### Checklist

- [X] I have read the documentation.
- [X] I have tried my code with the latest version of Node.js and Play-dl.

**Comment by YuzuZensai** [comment]
I'm experiencing the same issue.

Setting the cookie works about half the time, and changing the IP address also seems to help, but the problem ultimately happens again. Without setting the cookie, it wouldn't work at all.

**Comment by aymendps** [comment]
My discord bot (discord.js and play-dl) straight up stopped playing audio, no errors thrown. Tried the cookie method and no success. Not sure if this is a play-dl issue, Youtube issue, or discord.js issue.

**Comment by NikitaShevchuk** [comment]
same issue here. it's not related to discord.js, I've tried to play audio using ytdl-core and it works

**Comment by aymendps** [comment]
Anyone has a speculation what could be the issue in the code?

**Comment by YuzuZensai** [comment]
I forked the repo and based on my quick debugging and testing, it seems like YouTube asks for sign-in if you're hit with IP blocking or cookie banning.

However, if you are not banned, the stream stalls without playing any audio because the stream URL throws a 403 forbidden error. 
That looks like YouTube is detecting/blocking the web format extractor and giving 403 for the stream URL, and the Android extractor present in the code also doesn't seem to work due to YouTube API changes.

I managed to revive the working audio stream URL by using iOS extractor. But, just so you know, this fix probably doesn't fix the sign-in required issue, it probably tied to YouTube's BotGuard experiment or IP/cookie banned

[Here what I need to patch in order to make it work.](https://github.com/YuzuZensai/play-dl-test/commit/2bfbfe6decd68261747ba55800319f9906f12b03)

Basically I just disable the current web extractor, and change the android extractor to iOS instead.
I'm currently using this patch for my Discord bot and it seem to be working again, have to see....

EDIT: Updated the link, I renamed the fork

**Comment by startotoro** [comment]
any way to unban/bypass the sign in? my server have static ip but using the changes above worked on my local machine and the audio is now playing again

temp fix found, it seem to not block residential ip so i setup vpn from my vps to connect and route the traffic to it and it work. 

I currently am using npm install from git repo and cd into it and running npm run build, and doing that in a dockerfile when building my bot. Is there a better way or i have to wait for official fix?

**Comment by vqlion** [comment]
Thank you for the patch @YuzuZensai :heart_hands:
Currently using it, I'll also keep this thread updated on whether it keeps working or not.

**Comment by Noxde** [comment]
Thanks a lot for the patch @YuzuZensai

**Comment by alvarosdev** [comment]
@YuzuZensai Thanks!

**Comment by adamk33n3r** [comment]
I am now getting hit with this issue again. It looks to be happening at this request, and not the previous ones so it silently fails and was confusing to track down. https://github.com/Yoduh/play-dl-test/blob/main/play-dl/YouTube/utils/extractor.ts#L693

**Comment by Mauripache** [comment]
> I am now getting hit with this issue again. It looks to be happening at this request, and not the previous ones so it silently fails and was confusing to track down. https://github.com/Yoduh/play-dl-test/blob/main/play-dl/YouTube/utils/extractor.ts#L693

I am also hitting that same login required issue again, I forked from @YuzuZensai but it stopped working now, tried changing the client but I'm met with the same issue.

**Comment by adamk33n3r** [comment]
> > I am now getting hit with this issue again. It looks to be happening at this request, and not the previous ones so it silently fails and was confusing to track down. https://github.com/Yoduh/play-dl-test/blob/main/play-dl/YouTube/utils/extractor.ts#L693
> 
> I am also hitting that same login required issue again, I forked from [@YuzuZensai](https://github.com/YuzuZensai) but it stopped working now, tried changing the client but I'm met with the same issue.

I found success in switching to ytdl-core to fetch the stream by following what this other user did https://github.com/eritislami/evobot/pull/1645

**Comment by Mauripache** [comment]
> > > I am now getting hit with this issue again. It looks to be happening at this request, and not the previous ones so it silently fails and was confusing to track down. https://github.com/Yoduh/play-dl-test/blob/main/play-dl/YouTube/utils/extractor.ts#L693
> > 
> > 
> > I am also hitting that same login required issue again, I forked from [@YuzuZensai](https://github.com/YuzuZensai) but it stopped working now, tried changing the client but I'm met with the same issue.
> 
> I found success in switching to ytdl-core to fetch the stream by following what this other user did [eritislami/evobot#1645](https://github.com/eritislami/evobot/pull/1645)

Thanks @adamk33n3r ! that worked

**Comment by ZNGAS** [comment]
> > > > 我現在又遇到了這個問題。它似乎是在這個請求中發生的，而不是以前的請求，因此它悄無聲息地失敗並且難以追蹤。https://github.com/Yoduh/play-dl-test/blob/main/play-dl/YouTube/utils/extractor.ts#L693
> > > 
> > > 
> > > 我也再次遇到了同樣的需要登錄的問題，我從 [@YuzuZensai](https://github.com/YuzuZensai) 分叉，但它現在停止工作，嘗試更改用戶端，但我遇到了同樣的問題。
> > 
> > 
> > 我發現按照這個其他使用者所做的切換到 ytdl-core 來獲取流是成功的 [eritislami/evobot#1645](https://github.com/eritislami/evobot/pull/1645)
> 
> 謝謝 ！這奏效了

Could you tell me where to change?

**Comment by Mauripache** [comment]
> Could you tell me where to change?

Install @distube/ytdl-core and change where you use the stream method from play-dl to use the ytdl method from that library

**Comment by ZNGAS** [comment]
> > 您能告訴我在哪裡進行更改嗎？
> 
> 安裝 @distube/ytdl-core，並將使用 stream方法的位置從 play-dl 更改為使用該庫中的 ytdl 方法

Thank you!
